### PR TITLE
add syncdreamer as a image-to-3d pipeline

### PR DESCRIPTION
**Pull request by hitsz-zuoqi**
1、Syncdreamer is a image-to-3d method. It can generate 16 views based on a single image.
2、The added components have passed the unitest and ready for integration.
```

from modelscope.pipelines import pipeline
from modelscope.utils.constant import Tasks
import numpy as np
from PIL import Image

image23D = pipeline(
            Tasks.image_to_3d,
            model='Damo_XR_Lab/Syncdreamer')

result = image23D('./basketball.png')
# print(result)

np_content = []
for idx,img in enumerate(result['MViews']):
    np_content.append(np.array(result['MViews'][idx]))

np_content = np.concatenate(np_content, axis=1)
Image.fromarray(np_content).save("./demo_output/concat.png")
    # result['MViews'][idx].save("./demo_output/test_{:04d}.png".format(idx))

```

### CODE REVIEW

**Review by wenmengzhou — CHANGES_REQUESTED** [review]
remove directory `demo_output`

**Comment by hitsz-zuoqi** [comment]
> remove directory `demo_output`

ok, I will remove the demo_output

**Comment by hitsz-zuoqi** [comment]
> remove directory `demo_output`

I have removed the `demo_output` as requested. I think a potential modification you may request is add specific dependency to the `cv.txt`. The dependency I installed after I `install -r requirements/framework.txt` includes:
```

pytorch_lightning
kornia
matplotlib
git+https://github.com/openai/CLIP.git
omegaconf
taming-transformers-rom1504 
rembg

```

I don't know if I should add these requirements? plz.

**Comment by hitsz-zuoqi** [comment]
> > remove directory `demo_output`
> 
> I have removed the `demo_output` as requested. I think a potential modification you may request is add specific dependency to the `cv.txt`. The dependency I installed after I `install -r requirements/framework.txt` includes:
> 
> ```
> 
> pytorch_lightning
> kornia
> matplotlib
> git+https://github.com/openai/CLIP.git
> omegaconf
> taming-transformers-rom1504 
> rembg
> ```
> 
> I don't know if I should add these requirements? plz.

as your command, I have added these requirements as a requirements.txt in modelhub. Check it here:
https://modelscope.cn/models/Damo_XR_Lab/Syncdreamer/summary

File: modelscope/models/cv/image_to_3d/ldm/models/diffusion/sync_dreamer.py [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+        volume_xyz_ = volume_xyz_.permute(0, 2, 3, 4, 1)  # B*TN,D,H,W,3
+        spatial_volume_ = spatial_volume.unsqueeze(1).repeat(1, TN, 1, 1, 1, 1).view(B * TN, -1, V, V, V)
+        volume_feats = F.grid_sample(spatial_volume_, volume_xyz_, mode='bilinear', padding_mode='zeros', align_corners=True) # B*TN,C,D,H,W
+
+        v_embed_ = v_embed[torch.arange(B)[:,None], target_indices.view(B,TN)].view(B*TN, -1) # B*TN
+        t_embed_ = t_embed.unsqueeze(1).repeat(1,TN,1).view(B*TN,-1)
+        volume_feats_dict = self.frustum_volume_feats(volume_feats, t_embed_, v_embed_)
+        return volume_feats_dict, volume_depth
+
+class SyncMultiviewDiffusion(pl.LightningModule):
```
[the remark this answers is not in the document]
Comment (hitsz-zuoqi) [reply modelscope/models/cv/image_to_3d/ldm/models/diffusion/sync_dreamer.py:227]: > give some introduction of model here , and the paper link if it has

Done. I hide the install of clip using "os.system()", too.(In pipeline)

```

import os
os.system("pip install git+https://github.com/openai/CLIP.git")

```

[the remark this answers is not in the document]
Comment (hitsz-zuoqi) [reply modelscope/models/cv/image_to_3d/ldm/models/diffusion/sync_dreamer.py:227]: > give some introduction of model here , and the paper link if it has

Done.

**Review by wenmengzhou — COMMENTED** [review]
LGTM

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**